docs/whirlpool/docs/03-Whirlpools SDKs/01-Whirlpools/06-Send Transaction.mdx

```diff
@@ -0,0 +1,372 @@
+      "@solana/web3.js": "^2.0.0",
+      "tsx": "^4.19.2",
+      "typescript": "^5.7.2"
+    },
+    ```
+    ```tsx title="sendTransaction.ts"
+    import { openPositionInstructions, setWhirlpoolsConfig } from '@orca-so/whirlpools';
+    import { createSolanaRpc, address, pipe, createTransactionMessage, setTransactionMessageFeePayer, setTransactionMessageLifetimeUsingBlockhash, appendTransactionMessageInstructions, getComputeUnitEstimateForTransactionMessageFactory, prependTransactionMessageInstructions, signTransactionMessageWithSigners, getBase64EncodedWireTransaction, setTransactionMessageFeePayerSigner } from '@solana/web3.js';
+    import { getSetComputeUnitLimitInstruction, getSetComputeUnitPriceInstruction } from '@solana-program/compute-budget';
+    ```
+  </TabItem>
+  <TabItem value="rust" label="Rust">
+    ```toml title="Cargo.toml"
+    orca_whirlpools_client = "^1"
+    orca_whirlpools_core = "^1"
+    orca_whirlpools = "^1"
+    serde_json = { version = "^1.0" }
+    solana-client = { version = "^1.18" }
+    solana-sdk = { version = "^1.18" }
+    tokio = { version = "^1.41.1" }
+    ```
+    ```rust title="main.rs"
+    use orca_whirlpools::{
+      open_position_instructions, set_whirlpools_config_address, IncreaseLiquidityParam,
+      WhirlpoolsConfigInput,
+    };
+    use solana_client::nonblocking::rpc_client::RpcClient;
+    use solana_client::rpc_config::RpcSendTransactionConfig;
+    use solana_sdk::commitment_config::CommitmentLevel;
+    use solana_sdk::compute_budget::ComputeBudgetInstruction;
+    use solana_sdk::message::Message;
+    use solana_sdk::pubkey::Pubkey;
+    use solana_sdk::signature::Signature;
+    use solana_sdk::transaction::Transaction;
+    use solana_sdk::{signature::Keypair, signer::Signer};
+    use std::fs;
+    use std::str::FromStr;
+    use tokio::time::{sleep, Duration, Instant};
+    ```
+  </TabItem>
+</Tabs>
+
+
+### 1: Configure RPC Client and SDK
+To reliably land transactions, avoid using publicly available RPC nodes such as https://api.mainnet-beta.solana.com. Instead, consider using an RPC provider like Helius, Titan, or others. Many providers offer free tiers, which can be sufficient for testing purposes. You may need to adjust the retry mechanism in your code, specifically the waiting time between retries, to accommodate rate limits.
+
+<Tabs groupId="programming-languages">
+  <TabItem value="ts" label="Typescript" default>
+    ```tsx title="sendTransaction.ts"
+    const rpc = createSolanaRpc("<RPC_URL>");
+    await setWhirlpoolsConfig("solanaMainnet");
+    ```
+  </TabItem>
+  <TabItem value="rust" label="Rust">
+    ```rust title="main.rs"
+    #[tokio::main]
+    async fn main() {
+      let rpc = RpcClient::new("<RPC_URL>".to_string());
+      set_whirlpools_config_address(WhirlpoolsConfigInput::SolanaMainnet).unwrap();
+      // Rest of the code
+    }
+    ```
+  </TabItem>
+</Tabs>
+
+
+#### What is WhirlpoolsConfig?
```

Comment:
This feels kinda weird here. We are talking about sending and landing transactions. Then all of a sudden there is an explanation about what whirlpoolsconfig is